[cli] raster create - support resolution and % size (#14854)
OSGeo/gdal: doc/source/programs/gdal_raster_create.rst

    to respectively mean the special values not-a-number, positive infinity and
    minus infinity.

.. option:: --size <xsize[%]>,<ysize[%]>

    Set the size of the output file in pixels. First value is width. Second one
    is height. If width or height is set to 0,
    the other dimension will be guessed from the resolution implied by the
    bounding box.
    Values can be expressed as percentages of the size of the :option:`--like` dataset,
    if specified.

.. option:: --resolution <xres>,<yres>

>     Set the pixel resolution of the output file in CRS units. First value is

Reviewer: ```suggestion
    .. versionadded:: 3.14
    
    Set the pixel resolution of the output file in CRS units. First value is
```

>     width. Second one is height. If ``xres`` or ``yres`` is set to 0,

Reviewer: maybe?

```suggestion
    horizontal resolution. Second one is vertical resolution. If ``xres`` or ``yres`` is set to 0,
```

    the other dimension will be guessed from the size implied by the bounding box.

> 

Reviewer 2: I guess that what happens is something like `... the given non-zero resolution will be applied to both dimensions`

   .. code-block:: bash

      gdal raster create --like prototype.tif output.tif

.. example::
   :title: Initialize a blank GeoTIFF file from an input one, but with a different size and CRS

   .. code-block:: bash

>       gdal raster create --like prototype.tif --size=100,100 --crs=EPSG:3857 output.tif

Reviewer: you would typically use that while respecting the original resolution ratio, hence using a 0 value 

```suggestion
      gdal raster create --like prototype.tif --size=100,0 --crs=EPSG:3857 output.tif
```

.. example::
   :title: Initialize a blank GeoTIFF file from a bounding box and resolution

   .. code-block:: bash

      gdal raster create --bbox=2,49,3,50 --resolution=0.01,0.01 output.tif

.. example::
   :title: Initialize a blank GeoTIFF file from an input one with a different resolution

   .. code-block:: bash

>       gdal raster create --like prototype.tif --resolution=0.01,0.01 output.tif

Reviewer 2: The "like" source may have 0.01 as native resolution and the example would not match the description exactly. --resolution=50%,50% would certainly change the resolution.

Author: % was not implemented for resolution.  I will try to add it.

Reviewer 2: Ah, I forgot to consider that. Not sure if i% resolution is necessary because the same thing can be done with % size, and it may be confusing to remember, that 50% size means 200% resolution. And I think that nitpicking with the description can be skipped as well.
